### ISSUE TITLE
bug: Can't Unmarshal(b, v) if b is CBOR null (0xF6) or CBOR undefined (0xF7) and v is of type **time.Time

### ISSUE THREAD

**Issue by bangzek** [fxamacker/cbor#245]
### What version of fxamacker/cbor are you using?
2.2.0

### Does this issue reproduce with the latest release?
Yes

### What OS and CPU architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/zakaria/.cache/go-build"
GOENV="/home/zakaria/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/zakaria/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build661418315=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

https://play.golang.org/p/h4ws0o5srvA

### What did you expect to see?

I need to store nil time cause I need to store stop time, when stop time is nil that's mean the program is still running.

### What did you see instead?

panic

**Comment by bangzek** [comment]
Currently I work around it using zero time

**Comment by fxamacker** [comment]
Thanks for reporting this.  I'm looking into it.

**Comment by x448** [comment]
@fxamacker this scenario should be handled the same way as encoding/json.  

Consider changing title to something like: "bug: Can't Unmarshal(*time.Time) if *time.Time is a pointer to a nil pointer".

**Comment by fxamacker** [comment]
This problem with `Unmarshal(b, v)` happens when:

b is a CBOR null (0xF6) or CBOR undefined (0xF7)
&nbsp;  AND
v is a pointer to pointer of time.Time

A workaround is to use *time.Time instead of **time.Time.  For example, this works:

```
package main

import (
	"fmt"
	"time"

	"github.com/fxamacker/cbor/v2"
)

func main() {
	var t time.Time
	err := cbor.Unmarshal([]byte{0xf6}, &t)
	if err != nil {
		fmt.Printf("err: %s\n", err)
		return
	}
	fmt.Printf("t: %s\n", t)
}
```

**Comment by fxamacker** [comment]
Quick update on my schedule...I added the "in progress" tag on Feb 11, then Winter Storm Uri started damaging my place.  Repairs are still ongoing and it is not yet deemed "inhabitable" by my insurance company.

My apologies for the delay.

**Comment by fxamacker** [comment]
This issue was already fixed in commit c5db62bc082192ba32ef041d01e465ea224e2f01 in November 18, 2020.

Sorry about the confusion :smile: , closing as fixed.
